File: euler73.py
def gcd(a,b):
    r = a%b
    while r!=0:
        a =b
        b=r
        r=a%b
    return b

def between(ta, tb):
    a = ta[0]+tb[0]
    b = ta[1]+tb[1]
    g = gcd(a,b)
    a = int(a/g)
    b = int(b/g)
    return (a,b)

def dec(a):
 return a[0]/a[1]


def count(ta, tb, d):
    global s
    mid= between(ta, tb)
    if mid[1]<=d:
        s+=1
    else: return
    count(ta,mid,d)
    count(mid,tb,d)

ta=(1,3)
tb=(1,2)
d = 12000
s = 0
current= [ta,tb]
while len(current)!=0:
    tb = current.pop()
    ta = current.pop()
    mid = between(ta,tb)
    if mid[1] <=d:
        s+=1
        current.append(ta)
        current.append(mid)
        current.append(mid)
        current.append(tb)
print(s)

File: euler73.cpp
#include <iostream>
#include <vector>
using namespace std;

int gcd(int a, int b)
{
    int r = a%b;
    while(r!=0)
    {
        a = b;
        b = r;
        r = a%b;
    }
    return b;
}

vector<int> between(vector<int>a, vector<int>b)
{
    int aa = a[0]+b[0],
        bb = a[1]+b[1];
    vector<int> ret(2,0);
    int gc = gcd(aa,bb);
    ret[0] = aa/gc;
    ret[1] = bb/gc;
    return ret;
}

int main()
{
    int cnt =0;
    int d = 12000;
    vector< vector<int> > current;
    vector<int> start,end,mid;
    start.push_back(1);start.push_back(3);
    end.push_back(1);end.push_back(2);
    current.push_back(start);
    current.push_back(end);

    while(current.size()!=0)
    {
        end = current[current.size()-1];
        current.pop_back();
        start = current[current.size()-1];
        current.pop_back();
        mid = between(start, end);
        if(mid[1]<=d)
        {
            cnt+=1;
            current.push_back(start);
            current.push_back(mid);
            current.push_back(mid);
            current.push_back(end);
        }
    }
    cout << cnt;
}
